src/main/kotlin/pt/up/fe/ni/website/backend/service/AuthService.kt

```diff
@@ -49,10 +48,10 @@ class AuthService(
             try {
                 jwtDecoder.decode(refreshToken)
             } catch (e: Exception) {
-                throw ResponseStatusException(HttpStatus.UNAUTHORIZED, "invalid refresh token")
+                throw InvalidBearerTokenException("invalid refresh token")
             }
         if (jwt.expiresAt?.isBefore(Instant.now()) != false) {
-            throw ResponseStatusException(HttpStatus.UNAUTHORIZED, "refresh token has expired")
+            throw InvalidBearerTokenException("refresh token has expired")
```

Comment:
I didn't test this on my own so I'm going to ask. I like the change but is this exception being handled or does it just return an unexpected error?

Reply:
Yes, it is an AuthorizationException. This is being tested, too